static/app/components/commandPalette/types.tsx

```diff
@@ -0,0 +1,32 @@
+import type {ReactNode} from 'react';
+import type {LocationDescriptor} from 'history';
+
+export type CommandPaletteGroupKey = 'navigate' | 'add' | 'help';
+
+export type CommandPaletteAction = {
+  display: {
+    /** Primary text shown to the user */
+    label: string;
+    /** Additional context or description */
+    details?: string;
+    /** Icon to render for this action */
+    icon?: ReactNode;
+  };
+  /** Unique identifier for this action */
+  key: string;
+  /** Nested actions to show when this action is selected */
+  actions?: CommandPaletteAction[];
+  /** Section to group the action in the palette */
+  groupingKey?: CommandPaletteGroupKey;
+  /** Whether this action should be hidden from the palette */
+  hidden?: boolean;
```

Comment:
When would you want to use this vs just not provide the action?

Reply:
yeah this is very nice, we should keep this

Reply:
IMO it avoids ugly `...(shouldAddAction ? [{actionDef: ...}] : [])`

Reply:
True, but unless you make `hidden` a required prop, folks will never learn know it exists, and could still reach for the the ternary pattern anyways, so you just end up with two ways of doing it. It may also happen that some folks will know about hidden, but end up preferring to use the ternary anyways, and you can't really stop that.

An alternative that I can think of would be to have a `disabled:  string` prop which disables the action with an explanation. This could be more helpful than hiding it, and would enable a pattern where we explain why users don't have a permission to execute certain actions as opposed to hiding them (though this can't be enforced either, but it would be a way to provide a helpful explanation)